diff --git a/CMakeLists.txt b/CMakeLists.txt
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -340,8 +340,9 @@
 # it's not done here. (xz has translations, xzdec doesn't.)
 set(XZ_NLS_DEFAULT OFF)
 find_package(Intl)
 find_package(Gettext)
+find_package(prefix-stub CONFIG REQUIRED)
 if(Intl_FOUND AND GETTEXT_FOUND)
     set(XZ_NLS_DEFAULT ON)
 endif()
 
@@ -2080,14 +2081,15 @@
 
     # NOTE: The translations are in the "xz" domain and the .mo files are
     # installed as part of the "xz" target.
     if(XZ_NLS)
-        target_link_libraries(lzmainfo PRIVATE Intl::Intl)
+        target_link_libraries(lzmainfo PRIVATE Intl::Intl me::prefix-stub)
 
         target_compile_definitions(lzmainfo PRIVATE
                 ENABLE_NLS
                 PACKAGE="${TRANSLATION_DOMAIN}"
                 LOCALEDIR="${LOCALEDIR_DEFINITION}"
+                INSTALLDIR="${CMAKE_INSTALL_PREFIX}"
         )
     endif()
 
     install(TARGETS lzmainfo
@@ -2261,14 +2263,15 @@
         endif()
     endif()
 
     if(XZ_NLS)
-        target_link_libraries(xz PRIVATE Intl::Intl)
+        target_link_libraries(xz PRIVATE Intl::Intl me::prefix-stub)
 
         target_compile_definitions(xz PRIVATE
                 ENABLE_NLS
                 PACKAGE="${TRANSLATION_DOMAIN}"
                 LOCALEDIR="${LOCALEDIR_DEFINITION}"
+                INSTALLDIR="${CMAKE_INSTALL_PREFIX}"
         )
 
         file(STRINGS po/LINGUAS LINGUAS)
 
diff --git a/src/common/tuklib_gettext.h b/src/common/tuklib_gettext.h
--- src/common/tuklib_gettext.h
+++ src/common/tuklib_gettext.h
@@ -12,8 +12,9 @@
 #ifndef TUKLIB_GETTEXT_H
 #define TUKLIB_GETTEXT_H
 
 #include "tuklib_common.h"
+#include <prefix-stub/relocate.h>
 #include <locale.h>
 
 #ifndef TUKLIB_GETTEXT
 #	ifdef ENABLE_NLS
@@ -26,10 +27,10 @@
 #if TUKLIB_GETTEXT
 #	include <libintl.h>
 #	define tuklib_gettext_init(package, localedir) \
 		do { \
-			setlocale(LC_ALL, ""); \
-			bindtextdomain(package, localedir); \
+			setlocale(LC_ALL, ""); prefix_stub_set_orig_prefix(INSTALLDIR); \
+			bindtextdomain(package, prefix_stub_relocate(localedir)); \
 			textdomain(package); \
 		} while (0)
 #	define _(msgid) gettext(msgid)
 #else
