diff --git a/CMakeLists.txt b/CMakeLists.txt
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -206,9 +206,9 @@
 else()
     # Elsewhere a space is safe. This also keeps things compatible with
     # EBCDIC in case CMake-based build is ever done on such a system.
     set(PACKAGE_NAME_DEFINITION "${PACKAGE_NAME}")
-    set(LOCALEDIR_DEFINITION "${CMAKE_INSTALL_FULL_LOCALEDIR}")
+    set(LOCALEDIR_DEFINITION "${CMAKE_INSTALL_LOCALEDIR}")
 endif()
 
 # When used with MSVC, CMake can merge .manifest files with
 # linker-generated manifests and embed the result in an executable.
@@ -340,8 +340,9 @@
 # it's not done here. (xz has translations, xzdec doesn't.)
 set(XZ_NLS_DEFAULT OFF)
 find_package(Intl)
 find_package(Gettext)
+find_package(prefix-stub REQUIRED)
 if(Intl_FOUND AND GETTEXT_FOUND)
     set(XZ_NLS_DEFAULT ON)
 endif()
 
@@ -2056,9 +2057,9 @@
         src/common
         src/liblzma/api
     )
 
-    target_link_libraries(lzmainfo PRIVATE liblzma libgnu)
+    target_link_libraries(lzmainfo PRIVATE liblzma libgnu me::prefix-stub)
 
     if(WIN32 OR CYGWIN)
         # Add the Windows resource file for lzmainfo.exe.
         target_sources(lzmainfo PRIVATE src/lzmainfo/lzmainfo_w32res.rc)
@@ -2158,9 +2159,9 @@
             src/xz/list.h
         )
     endif()
 
-    target_link_libraries(xz PRIVATE liblzma libgnu)
+    target_link_libraries(xz PRIVATE liblzma libgnu me::prefix-stub)
 
     if(USE_POSIX_THREADS)
         # src/xz/signals.c uses mythread_sigmask() which with POSIX
         # threads calls pthread_sigmask(). Thus, we need the threading
diff --git a/src/common/tuklib_gettext.h b/src/common/tuklib_gettext.h
--- src/common/tuklib_gettext.h
+++ src/common/tuklib_gettext.h
@@ -12,8 +12,9 @@
 #ifndef TUKLIB_GETTEXT_H
 #define TUKLIB_GETTEXT_H
 
 #include "tuklib_common.h"
+#include <prefix-stub/relocate.h>
 #include <locale.h>
 
 #ifndef TUKLIB_GETTEXT
 #	ifdef ENABLE_NLS
@@ -27,9 +28,9 @@
 #	include <libintl.h>
 #	define tuklib_gettext_init(package, localedir) \
 		do { \
 			setlocale(LC_ALL, ""); \
-			bindtextdomain(package, localedir); \
+			bindtextdomain(package, prefix_stub_relocate(localedir)); \
 			textdomain(package); \
 		} while (0)
 #	define _(msgid) gettext(msgid)
 #else
