diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..42bb970
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,106 @@
+cmake_minimum_required(VERSION 4.1)
+
+project(rtmpdump VERSION 2.6 LANGUAGES C)
+
+include(GNUInstallDirs)
+include(GenerateExportHeader)
+
+set(CMAKE_C_STANDARD 17)
+set(BUILD_SHARED_LIBS ON)
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+find_package(getopt CONFIG REQUIRED)
+find_package(ZLIB REQUIRED)
+find_package(OpenSSL REQUIRED)
+
+add_library(rtmp
+  librtmp/amf.c
+  librtmp/hashswf.c
+  librtmp/log.c
+  librtmp/parseurl.c
+  librtmp/rtmp.c
+)
+target_link_libraries(rtmp
+    PRIVATE
+      OpenSSL::Crypto
+      OpenSSL::SSL
+      ZLIB::ZLIB
+      WS2_32
+      WinMM
+)
+target_compile_definitions(rtmp
+  PRIVATE
+    _CRT_DECLARE_NONSTDC_NAMES=1
+    USE_OPENSSL
+    OPENSSL_API_COMPAT=0x10100000L
+)
+generate_export_header(rtmp EXPORT_FILE_NAME librtmp/rtmp_export.h)
+target_sources(
+  rtmp
+  PUBLIC
+    FILE_SET HEADERS
+    BASE_DIRS
+      $<TARGET_PROPERTY:rtmp,BINARY_DIR>
+      ${CMAKE_CURRENT_LIST_DIR}
+      FILES
+        librtmp/amf.h
+        librtmp/bytes.h
+        librtmp/dh.h
+        librtmp/dhgroups.h
+        librtmp/handshake.h
+        librtmp/http.h
+        librtmp/log.h
+        librtmp/rtmp_sys.h
+        librtmp/rtmp.h
+        ${CMAKE_CURRENT_BINARY_DIR}/librtmp/rtmp_export.h
+)
+
+add_compile_definitions(RTMPDUMP_VERSION=\"v${PROJECT_VERSION}\")
+
+add_executable(rtmpdump rtmpdump.c)
+target_link_libraries(rtmpdump PRIVATE getopt::getopt_shared rtmp WS2_32)
+
+add_executable(rtmpsrv rtmpsrv.c thread.c)
+target_link_libraries(rtmpsrv PRIVATE getopt::getopt_shared rtmp WS2_32)
+
+add_executable(rtmpsuck rtmpsuck.c thread.c)
+target_link_libraries(rtmpsuck PRIVATE getopt::getopt_shared rtmp WS2_32)
+
+add_executable(rtmpgw rtmpgw.c thread.c)
+target_link_libraries(rtmpgw PRIVATE getopt::getopt_shared rtmp WS2_32)
+
+install(TARGETS
+    rtmp
+    rtmpdump
+    rtmpsrv
+    rtmpsuck
+    rtmpgw
+)
+install(
+  TARGETS rtmp
+  FILE_SET HEADERS
+  FILE_SET generated_headers
+)
+install(
+  FILES
+    rtmpdump.1
+    DESTINATION ${CMAKE_INSTALL_MANDIR}/man1
+)
+install(FILES rtmpgw.8
+  DESTINATION ${CMAKE_INSTALL_MANDIR}/man8
+)
+install(FILES librtmp/librtmp.3
+  DESTINATION ${CMAKE_INSTALL_MANDIR}/man3
+)
+
+block()
+  set(prefix "${CMAKE_INSTALL_PREFIX}")
+  set(libdir "${CMAKE_INSTALL_LIBDIR}")
+  set(VERSION "${PROJECT_VERSION}")
+  set(CRYPTO_REQ "openssl, zlib")
+  set(PRIVATE_LIBS "-lWS2_32 -lWinMM")
+  configure_file(librtmp/librtmp.pc.in librtmp.pc @ONLY)
+endblock()
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/librtmp.pc
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
+)
diff --git a/librtmp/librtmp.pc.in b/librtmp/librtmp.pc.in
index 551b4d9..99fc3b0 100644
--- a/librtmp/librtmp.pc.in
+++ b/librtmp/librtmp.pc.in
@@ -1,6 +1,6 @@
 prefix=@prefix@
 exec_prefix=${prefix}
-libdir=@libdir@
+libdir=${prefix}/@libdir@
 incdir=${prefix}/include
 
 Name: librtmp
@@ -8,6 +8,6 @@ Description: RTMP implementation
 Version: @VERSION@
 Requires: @CRYPTO_REQ@
 URL: http://rtmpdump.mplayerhq.hu
-Libs: -L${libdir} -lrtmp -lz @PUBLIC_LIBS@
+Libs: -L${libdir} -lrtmp @PUBLIC_LIBS@
 Libs.private: @PRIVATE_LIBS@
 Cflags: -I${incdir}
diff --git a/librtmp/log.h b/librtmp/log.h
index 2adb111..0fd4334 100644
--- a/librtmp/log.h
+++ b/librtmp/log.h
@@ -27,6 +27,7 @@
 #include <stdio.h>
 #include <stdarg.h>
 #include <stdint.h>
+#include "librtmp//rtmp_export.h"
 
 #ifdef __cplusplus
 extern "C" {
@@ -43,7 +44,7 @@ typedef enum
   RTMP_LOGDEBUG, RTMP_LOGDEBUG2, RTMP_LOGALL
 } RTMP_LogLevel;
 
-extern RTMP_LogLevel RTMP_debuglevel;
+extern RTMP_EXPORT RTMP_LogLevel RTMP_debuglevel;
 
 typedef void (RTMP_LogCallback)(int level, const char *fmt, va_list);
 void RTMP_LogSetCallback(RTMP_LogCallback *cb);
diff --git a/librtmp/rtmp.h b/librtmp/rtmp.h
index 6d7dd89..9c5f2a3 100644
--- a/librtmp/rtmp.h
+++ b/librtmp/rtmp.h
@@ -34,6 +34,7 @@
 #include <stddef.h>
 
 #include "amf.h"
+#include "librtmp/rtmp_export.h"
 
 #ifdef __cplusplus
 extern "C"
@@ -65,9 +66,9 @@ extern "C"
 
 #define	RTMP_CHANNELS	65600
 
-  extern const char RTMPProtocolStringsLower[][7];
-  extern const AVal RTMP_DefaultFlashVer;
-  extern int RTMP_ctrlC;
+  extern RTMP_EXPORT const char RTMPProtocolStringsLower[][7];
+  extern RTMP_EXPORT const AVal RTMP_DefaultFlashVer;
+  extern RTMP_EXPORT int RTMP_ctrlC;
 
   uint32_t RTMP_GetTime(void);
 
diff --git a/librtmp/rtmp_sys.h b/librtmp/rtmp_sys.h
index 85d7e53..32e8555 100644
--- a/librtmp/rtmp_sys.h
+++ b/librtmp/rtmp_sys.h
@@ -28,16 +28,15 @@
 #include <ws2tcpip.h>
 
 #ifdef _MSC_VER	/* MSVC */
-#define snprintf _snprintf
 #define strcasecmp stricmp
 #define strncasecmp strnicmp
-#define vsnprintf _vsnprintf
+#define ftello64 _ftelli64
+#define fseeko64 _fseeki64
 #endif
 
 #define GetSockError()	WSAGetLastError()
 #define SetSockError(e)	WSASetLastError(e)
 #define setsockopt(a,b,c,d,e)	(setsockopt)(a,b,c,(const char *)d,(int)e)
-#define EWOULDBLOCK	WSAETIMEDOUT	/* we don't use nonblocking, but we do use timeouts */
 #define sleep(n)	Sleep(n*1000)
 #define msleep(n)	Sleep(n)
 #define SET_RCVTIMEO(tv,s)	int tv = s*1000
diff --git a/rtmpdump.c b/rtmpdump.c
index 13741a7..120eeaf 100644
--- a/rtmpdump.c
+++ b/rtmpdump.c
@@ -278,8 +278,8 @@ GetLastKeyframe(FILE * file,	// output file [in]
 		int *initialFrameType,	// initial frame type (audio/video) [out]
 		uint32_t * nInitialFrameSize)	// length of initialFrame [out]
 {
-  const size_t bufferSize = 16;
-  char buffer[bufferSize];
+  size_t bufferSize = 16;
+  char buffer[16];
   uint8_t dataType;
   int bAudioOnly;
   off_t size;
