diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
--- Configurations/10-main.conf
+++ Configurations/10-main.conf
@@ -1591,9 +1591,9 @@
         bin_lflags       => add("/subsystem:console /opt:ref"),
         ex_libs          => add(sub {
             my @ex_libs = ();
             push @ex_libs, 'ws2_32.lib' unless $disabled{sock};
-            push @ex_libs, 'gdi32.lib advapi32.lib crypt32.lib user32.lib';
+            push @ex_libs, 'gdi32.lib advapi32.lib crypt32.lib user32.lib prefix-stub.lib';
             return join(" ", @ex_libs);
         }),
     },
     "VC-WIN64-common" => {
diff --git a/crypto/defaults.c b/crypto/defaults.c
--- crypto/defaults.c
+++ crypto/defaults.c
@@ -8,8 +8,9 @@
  */
 
 #include <stdio.h>
 #include <openssl/opensslv.h>
+#include <prefix-stub/relocate.h>
 #include "internal/thread_once.h"
 #include "internal/cryptlib.h"
 #include "internal/e_os.h"
 
@@ -139,9 +140,9 @@
     if (!RUN_ONCE(&defaults_setup_init, do_defaults_setup))
         return NULL;
     return (const char *)openssldirptr;
 #else
-    return OPENSSLDIR;
+    return prefix_stub_relocate(OPENSSLDIR);
 #endif
 }
 
 /**
@@ -155,9 +156,9 @@
     if (!RUN_ONCE(&defaults_setup_init, do_defaults_setup))
         return NULL;
     return (const char *)modulesdirptr;
 #else
-    return MODULESDIR;
+    return prefix_stub_relocate(MODULESDIR);
 #endif
 }
 
 /**
diff --git a/crypto/dllmain.c b/crypto/dllmain.c
--- crypto/dllmain.c
+++ crypto/dllmain.c
@@ -13,8 +13,9 @@
 #if defined(_WIN32) || defined(__CYGWIN__)
 #ifdef __CYGWIN__
 /* pick DLL_[PROCESS|THREAD]_[ATTACH|DETACH] definitions */
 #include <windows.h>
+#include <prefix-stub/relocate.h>
 /*
  * this has side-effect of _WIN32 getting defined, which otherwise is
  * mutually exclusive with __CYGWIN__...
  */
@@ -29,8 +30,9 @@
 BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved)
 {
     switch (fdwReason) {
     case DLL_PROCESS_ATTACH:
+        prefix_stub_set_orig_prefix(INSTALLDIR);
         break;
     case DLL_THREAD_ATTACH:
         break;
     case DLL_THREAD_DETACH:
diff --git a/crypto/ct/ct_log.c b/crypto/ct/ct_log.c
--- crypto/ct/ct_log.c
+++ crypto/ct/ct_log.c
@@ -162,9 +162,9 @@
 {
     const char *fpath = ossl_safe_getenv(CTLOG_FILE_EVP);
 
     if (fpath == NULL)
-        fpath = CTLOG_FILE;
+        fpath = prefix_stub_relocate(CTLOG_FILE);
 
     return CTLOG_STORE_load_file(store, fpath);
 }
 
