diff --git a/m4/ax_gcc_x86_cpuid.m4 b/m4/ax_gcc_x86_cpuid.m4
--- m4/ax_gcc_x86_cpuid.m4
+++ m4/ax_gcc_x86_cpuid.m4
@@ -61,15 +61,13 @@
 [AC_REQUIRE([AC_PROG_CC])
 AC_LANG_PUSH([C])
 AC_CACHE_CHECK(for x86 cpuid $1 output, ax_cv_gcc_x86_cpuid_$1,
  [AC_RUN_IFELSE([AC_LANG_PROGRAM([#include <stdio.h>], [
-     int op = $1, eax, ebx, ecx, edx;
+     int ex[[4]];
+     __cpuid(ex, $1);
      FILE *f;
-      __asm__("cpuid"
-        : "=a" (eax), "=b" (ebx), "=c" (ecx), "=d" (edx)
-        : "a" (op));
      f = fopen("conftest_cpuid", "w"); if (!f) return 1;
-     fprintf(f, "%x:%x:%x:%x\n", eax, ebx, ecx, edx);
+     fprintf(f, "%x:%x:%x:%x\n", ex[[0]], ex[[1]], ex[[2]], ex[[3]]);
      fclose(f);
      return 0;
 ])],
      [ax_cv_gcc_x86_cpuid_$1=`cat conftest_cpuid`; rm -f conftest_cpuid],
diff --git a/configure.ac b/configure.ac
--- configure.ac
+++ configure.ac
@@ -34,9 +34,9 @@
 
 # Checks for library functions.
 AC_CHECK_FUNCS([memset])
 
-dnl AM_PROG_AR 
+AM_PROG_AR
 AC_PROG_LIBTOOL
 dnl LT_INIT
 
 
