diff --git a/llvm/include/llvm/ExecutionEngine/Orc/BacktraceTools.h b/llvm/include/llvm/ExecutionEngine/Orc/BacktraceTools.h
index 454d7654b66bf..c9366aef4de00 100644
--- a/llvm/include/llvm/ExecutionEngine/Orc/BacktraceTools.h
+++ b/llvm/include/llvm/ExecutionEngine/Orc/BacktraceTools.h
@@ -78,7 +78,7 @@ class LLVM_ABI DumpedSymbolTable {
 
   /// Given a backtrace, try to symbolicate any unsymbolicated lines using the
   /// symbol addresses in the dumped symbol table.
-  LLVM_ABI std::string symbolicate(StringRef Backtrace);
+  std::string symbolicate(StringRef Backtrace);
 
 private:
   DumpedSymbolTable(std::unique_ptr<MemoryBuffer> SymtabBuffer);
