diff --git a/src/tool_doswin.c b/src/tool_doswin.c
--- src/tool_doswin.c
+++ src/tool_doswin.c
@@ -22,8 +22,11 @@
  *
  ***************************************************************************/
 #include "tool_setup.h"
 
+#include <prefix-stub/relocate.h>
+#pragma comment(lib, "prefix-stub.lib")
+
 #if defined(_WIN32) || defined(MSDOS)
 
 #include "curlx/basename.h" /* for curlx_basename() */
 #include "curlx/version_win32.h" /* for curlx_verify_windows_version() */
@@ -562,8 +565,14 @@
 
   buf[0] = TEXT('\0');
 
   res_len = SearchPath(NULL, bundle_file, NULL, MAX_PATH, buf, &ptr);
+  if (res_len == 0) {
+    const TCHAR *p_ssl = prefix_stub_relocate_w(TEXT("SSL"));
+    if (*p_ssl) {
+      res_len = SearchPath(p_ssl, bundle_file, NULL, MAX_PATH, buf, &ptr);
+    }
+  }
   if(res_len > 0) {
     curlx_free(config->cacert);
     config->cacert = curlx_convert_tchar_to_UTF8(buf);
     if(!config->cacert)
diff --git a/src/tool_operate.c b/src/tool_operate.c
--- src/tool_operate.c
+++ src/tool_operate.c
@@ -2133,8 +2133,10 @@
     }
 #elif !defined(CURL_WINDOWS_UWP) && !defined(CURL_DISABLE_CA_SEARCH)
     result = FindWin32CACert(config, TEXT("curl-ca-bundle.crt"));
     if(result)
+      result = FindWin32CACert(config, TEXT("cacert.pem"));
+    if(result)
       goto fail;
 #endif
   }
 #endif
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
--- lib/CMakeLists.txt
+++ lib/CMakeLists.txt
@@ -65,9 +65,9 @@
   # Suffix implib name with "_imp" by default, to avoid conflicting with
   # the generated static "libcurl.lib" (typically with MSVC).
   if(WIN32 AND BUILD_SHARED_LIBS AND
      CMAKE_IMPORT_LIBRARY_SUFFIX STREQUAL CMAKE_STATIC_LIBRARY_SUFFIX)
-    set(IMPORT_LIB_SUFFIX "_imp")
+    set(IMPORT_LIB_SUFFIX "")
   else()
     set(IMPORT_LIB_SUFFIX "")
   endif()
 endif()
